### DISCUSSION TITLE
Version 2021.1127.0-lazer doesn't launch

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#15834] [Q&A] [answered]
Launched correctly after updating, then the second launch doesn't do anything.
I'm not coding literate enough to know what's going on here; this is the only log that is generated: [runtime.log.txt](https://github.com/ppy/osu/files/7611746/runtime.log.txt)
The update logs don't contain any errors, but I can provide them if necessary.

**Comment by bdach** [comment, marked as answer]
the log shows an error that seems that indicate that you may have downgraded the lazer version (realm is saying that the database schema version is lower than the expected one). have you done anything of the sort?

in any case a stop-gap fix for this is to remove `client.realm` from your game files manually. right now you should not lose data by doing this.

**Reply by zachmanthethird** [reply]
This solution downgraded me to 2021.1120.0-lazer and doesn't detect any newer versions. I assume that's intended.
I have never deliberately downgraded the lazer version; no; however, I was inactive during October, so I updated directly from 2021.927.0 to 2021.1120.0, so I'm not sure if that may also be the cause of my problems.

**Reply by bdach** [reply]
the solution i gave you did not "downgrade you to 2021.1120.0". your log you attached already shows that version.

```
2021-11-27 12:50:55 [verbose]: ----------------------------------------------------------
2021-11-27 12:50:55 [verbose]: runtime Log for zachm (LogLevel: Verbose)
2021-11-27 12:50:55 [verbose]: Running osu 2021.1120.0.0 on .NET 5.0.11
2021-11-27 12:50:55 [verbose]: Environment: Windows (Microsoft Windows NT 10.0.22000.0), 8 cores
2021-11-27 12:50:55 [verbose]: ----------------------------------------------------------
```

so either something went wrong with the update or it must be something on your machine.

**Reply by zachmanthethird** [reply]
I'm going to guess it's probably a mix of both? Here are all of the other logs from when it was running 2021.1127.0 just to show I'm not crazy...
[database.log.txt](https://github.com/ppy/osu/files/7611813/database.log.txt) [network.log.txt](https://github.com/ppy/osu/files/7611814/network.log.txt) [runtime.log.txt](https://github.com/ppy/osu/files/7611815/runtime.log.txt) [updater.log.txt](https://github.com/ppy/osu/files/7611816/updater.log.txt)
Should I attempt a re-install? I used the "Change folder location..." to move my installation to another drive (version 2021.1120.0)... is that causing the problem?

**Reply by bdach** [reply]
the other logs you attached were not overwritten from the past run. look at the timestamps, the `2021.1120.0` one starts at 12:50 UTC, the `2021.1127.0` all have previous dates.

this may be related to moving your files to another location. did you lose your imported beatmaps?

**Reply by zachmanthethird** [reply]
So I updated to 2021.1127.0 at about 12:25 UTC and the installation worked fine. Then, when relaunching the game at 12:50, the game wouldn't launch. I never lost any beatmaps/scores.

Update: so when 2021.1127.0 installed, it didn't change the osu!.exe in the root folder. The osu!.exe in the app-2021.1127.0 folder works fine. Not sure what the best way is to fix though, the two files don't look the same. Would a temporary shortcut work or should I reinstall just to get the correct exe file?

**Reply by bdach** [reply]
that sounds like a squirrel failure of some sort. you could try backing up the old executable and replacing it with the new one, but a full reinstall is likely safer.

**Reply by peppy** [reply]
The top level executable redirects to the correct version. That's the one you should be using.
